File: stdin_hello_world/stdin_hello_world.py
from sys import stdin

print("Enter input: ", end="", flush=True)

user_input = stdin.readline().rstrip()

print(user_input)

if user_input == 'hello':
    print("hi!")
elif user_input == 'world':
    print("hey!")
else:
    print("hello?")

File: stdin_hello_world/stdin_hello_world.rs
use std::io;
use std::io::Write;

fn main() {
    print!("Enter input: ");

    if let Err(e) = io::stdout().flush() {
        eprintln!("Error flushing to stdout: {:?}", e);
    }

    let mut input = String::new();

    io::stdin()
        .read_line(&mut input)
        .expect("Error reading from stdin");

    let input = input.trim_end();

    println!("Input: {}", input);

    match &input[..] {
        "hello" => println!("hi!"),
        "world" => println!("hey"),
        _ => println!("hello?"),
    }
}
